tsdb/head_append.go

```diff
@@ -1166,14 +1276,57 @@ func handleAppendableError(err error, appended, oooRejected, oobRejected, tooOld
 // operations on the series after appending the samples.
 //
 // There are also specific functions to commit histograms and float histograms.
-func (a *headAppender) commitSamples(acc *appenderCommitContext) {
+func (a *headAppender) commitFloats(b *appendBatch, acc *appenderCommitContext) {
 	var ok, chunkCreated bool
 	var series *memSeries
 
-	for i, s := range a.samples {
-		series = a.sampleSeries[i]
+	for i, s := range b.floats {
+		series = b.floatSeries[i]
 		series.Lock()
 
+		if value.IsStaleNaN(s.V) {
```

Comment:
I think you need the same check in commitHistograms and commitFloatHistograms.

If sum == staleNaN then check for last*Histogram and if both is true, revert to staleNaN float.